From 777f11b0ac97c5739a925ab50d248aa4039ae834 Mon Sep 17 00:00:00 2001
From: Geeta Chauhan <gchauhan@fb.com>
Date: Thu, 31 Oct 2019 09:14:44 -0700
Subject: [PATCH] Fix for torch.save not saving source files

Summary: Fixed the reference to correct object

Test Plan:
Verified by running the test_torch tests and a sample test case to see file getting saved with no errors or warnings
import torch
m = torch.nn.Linear(500, 10)
torch.save(m, './test.pth')

Reviewers: vincentqb

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]
---
 torch/serialization.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/torch/serialization.py b/torch/serialization.py
index ca5a43c7bd59..678fb17029e0 100644
--- a/torch/serialization.py
+++ b/torch/serialization.py
@@ -287,12 +287,13 @@ def persistent_id(obj):
             source_file = source = None
             try:
                 source_lines, _, source_file = get_source_lines_and_file(obj)
-                source = ''.join(obj)
+                source = ''.join(source_lines)
             except Exception:  # saving the source is optional, so we can ignore any errors
                 warnings.warn("Couldn't retrieve source code for container of "
                               "type " + obj.__name__ + ". It won't be checked "
                               "for correctness upon loading.")
             return ('module', obj, source_file, source)
+
         elif torch.is_storage(obj):
             storage_type = normalize_storage_type(type(obj))
             # Offset is always 0, but we keep it for backwards compatibility
